### DISCUSSION TITLE
swapping r-link evolved unit

### DISCUSSION THREAD

**Discussion by ShaneHash** [cedricp/ddt4all#1299] [General]
Hi All

I have just bought a 2016 Twingo III with a non-functional r-link - back screen, occasional TomTom boot screen, but no go. 

The local dealer charged me £100 to tell me the problem (hopefully) and they say £800 more will allow them to replace the 'multimedia interface computer', which I assume is the R-link touchscreen (ouch!).  

I have bought a second-hand touchscreen unit from a scrapped Twingo of the same year and similar spec which I want to swap in - hopefully, this will move things forward a bit without me spending £800.

I have an ILC reader and branched cable which I need to do the cable modification on, and need to sort out my laptop for either Pyren or DDT or both.

As I understand it, if I use DTT4All to make the new touchscreen unit VIN and CRC match the rest of the car, that should work. Unless there's something I haven't picked up on while reading everything I can find on-line.

I did the Pryenbot thing and have some macros based on the VIN number.

Question: Which is the best way to go forward - DDT4All or the Pyren macro? Does the Macro set the touchscreen settings to how they should be?

**Comment by AndyIreland** [comment]
when they say the 'multimedia interface computer' - they might be talking about the BIC ECU unit?  totally different from the Radio 2012 and MFD Touchscreen (TomTom rlink1) - you might have to fill out that BIC with details especially if someone has blanked it by pressing soft reset instead of hard reset , maybe if they were trying to put android auto on their rlink. - you might even have to do something special in the BIC to 'wake up' the radio and rlink touchscreen (MFD) to see them  in DDT4ALL / PyREN 

If you do change rlink to a second hand one , I wonder how you will get on with any 'anti theft' issues - you most certainly with a secondhand unit at the very least get the VIN number of the car the rlink came out from so that you can log onto my renault with the VIN number and get the 4 digit unlock code or give the VIN number of the car you have taken the rlink out from to get the 4 digit code off them to operate the rlink

Best way forward for the moment is , use your original touchscreen that came with your touchscreen - fire up PyREN, tick the CAN2 box - search for vehicle , then see if 3 ECU's start up - BIC, Radio2012, MFD xxx

**Reply by ShaneHash** [reply]
True enough - they might be something other than the touchscreen, and I guess they want any future business so I get that they wouldn't be precise about what they are proposing. My thinking was that they wanted to charge £800, and I saw a brand new touchscreen somewhere for £600 and I guessed that if they were charging about that for a touch screen and added a couple of hours of workshop time, that would be about right.

They did say that with their equipment they couldn't find any of the r-link connected computers running, so they thought the problem might be an internal fault with whatever module connects them all together.

The symptoms on the screen look like a boot-loop (I mess with phone ROMS, so that's something I have seen a few times!). It looks like it wants to start, but never actually does. There's no blue tooth connection I can find, and no reversing camera. 

But I will take your advice and stick with it as it is for now. I have the cable sorted out, and have ECU-Tweaker on my phone., so should be able to check that part tomorrow. I'm still struggling with the mobile version of Pyren. I'm short on time at the moment, so I might take a few days for me to get my laptop up and running, but the plan now is to double check that I can't get a connection to the R-link.

**Comment by AndyIreland** [comment]
you might find this handy, and might have to make these changes to your BIC ecu to wake up the power to your Radio ecu and MFD ecu - although the script on this BIC is for Clio, I think your one for Twingo will be different script number:
https://github.com/cedricp/ddt4all/issues/92#issuecomment-457928308

![image](https://github.com/user-attachments/assets/a524361b-def0-4d2a-8bc9-83969068b30e)

**Comment by ShaneHash** [comment]
Had another look today. I suspect the reader I have isn't working. There's a power light, but none of the other lights do anything. I'll try it on my other car tomorrow, and if it doesn't work, I'll send it back and try another brand. I'm thinking I need one that doesn't use bluetooth or wi-fi to connect - I'm sure I have seen a USB version somewhere.

Interestingly, while I was messing around with it and checking the bluetooth, I got 'R-Link' as a bluetooth 'connect to' device option. But it didn't stay long enough to connect to. I also got two different DMRRBA-004 option, both of which I could add as saved devices. 

ECU-tweaker said that any attempt to connect via bluetooth failed. EDIT: Version 1.9 will connect and hold a bluetooth connection.

**Comment by ShaneHash** [comment]
I'm still here. I have found a ODB2 with a USB connector. It's a Fordscan type so the cable modification is slightly different. Struggling to get Pyren to work. I have found an old installer that does seem to work - Pyren184. I need to work out how to update that. EDIT: Updated - I just needed to put the master files into the folder, replacing the old files. Should be able to try again on Friday.

**Comment by ShaneHash** [comment]
Still struggling to get anything to work.

I am satisfied that I have software that works, and the laptop seems to be able to handle everything in demo mode.

But I'm getting nowhere with connecting to the car and I am not sure if that is down to the car or something to do with the OBD2 box or the modfied wiring. Either way, no data is coming back from the car. Pyren doesn't connect at all.

This comes up in the logs:

>
>[12:12:03.453]021003
<[1.061]021003
NO DATA

>
>[12:12:04.514]0210C0
<[0.624]0210C0
NO DATA

>

Next step is to look at the wiring of the y-cable - maybe I have made a mistake there?

LOG: [elm_ddt.txt](https://github.com/user-attachments/files/18039776/elm_ddt.txt)


![Screenshot_1](https://github.com/user-attachments/assets/94a8c410-0409-4833-8ec2-b16a715a4261)

![Screenshot_5](https://github.com/user-attachments/assets/16d24dcb-e3cf-4cd5-af5c-4e404f7d2164)

**Comment by ShaneHash** [comment]
Wiring of the Y-cable...

![odb2v2jpg](https://github.com/user-attachments/assets/3b211288-39aa-4da9-9908-336fe2701721)

This is what it should look like, I think, and is how it is actually set.

13 and 3 swapped, and 12 and 11 swapped for CAN Bus.

So I think the cable is OK and so is the ODB2 box.

**Comment by ShaneHash** [comment]
Still working on this, The wet and cold and Xmas has slowed things down.

Yesterday, for some reason my R-link decided to do something unexpected. It booted, twice, but only briefly. It went from the tomtom logo, to the purple r-link amination, then asked for the radio code. But before I could do anything, it died again. On the second occasion it got past the radio code screen to a screen with  --:--, then died again, then back to the normal failure to boot.

I'm now thinking I need to check the various connections related to r-link for a dodgy cable connection at the back of the touchscreen or at the BIC. I'm not sure where the BIC is on a Twingo 3, but there aren't many places it could be at, to be fair. It's a small car.

**Comment by ShaneHash** [comment]
OK - the weather has improved a bit so I have resumed working on this.

I have had the touchscreen light up to the 'enter radio code' a couple of times now, but only when the cabin is very warm. Given this, I don't think it's a wrong script thing on the BIC.

I tried a second hand touchscreen on a 30-day return, thinking that even if I get the anti-theft screen that would show the touch screen is the problem. The second hand touchscreen behaves exactly as the original one, so I have put the old one back. Either both touch screen have failed in exactly the same way, or it is something else.

I found the BIC behind the glove box - I needed to remove the glove box to get to it. Unplugging it and re-plugging it in didn't make any difference, so I found a BIC specifically from a Twingo on Ebay, which I have ordered from a seller in Lithuania. I'm hopeful the problem is the BIC and this is the part Renault said they would fit for me for a huge amount of money. Again, if I get any change I'll try and work with that. What I have now is not working.

**Comment by ShaneHash** [comment]
New BIC allows me to get to the "enter Radio PIN." screen. But the number I have for it via 'My Renault' app doesn't work.

But this feels like a step forward, at least...

**Reply by ShaneHash** [reply]
Just to end this: Turns out the replacement BIC was for a Clio V, not a Twingo (based in the VIN number programmed), but it still worked. I discovered the Radio had been swapped at some point and I needed a different PIN code. I removed the Radio to get the security code which allowed me to get the PIN number. Put the PN number in and everything started working as it should do.
